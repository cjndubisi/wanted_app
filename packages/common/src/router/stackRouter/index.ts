import { Platform } from 'react-native';

const native = Platform.select({
  ios: () => require('./stackRouter.native'),
  android: () => require('./stackRouter.native'),
  default: () => require('./stackRouter.web'),
})();

console.log('nansns', native);
export default native;
