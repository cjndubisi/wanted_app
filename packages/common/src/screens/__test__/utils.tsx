import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import { BaseNavigationContainer } from '@react-navigation/core';
import { fireEvent, render } from '@testing-library/react-native';
import { AuthProvider } from '../../context/AuthContext';

export const withNavigation = ({
  screens = {},
}: {
  screens: { [component: string]: { component: any; path: string } };
}) => {
  return class extends React.Component {
    render() {
      const Stack = createStackNavigator();

      return (
        <BaseNavigationContainer>
          <Stack.Navigator>
            {Object.keys(screens).map(name => (
              <Stack.Screen
                key={name}
                name={screens[name].path}
                component={screens[name].component}
              />
            ))}
          </Stack.Navigator>
        </BaseNavigationContainer>
      );
    }
  };
};

export const withProviders = (Component: any) => {
  return class extends React.Component {
    render() {
      return (
        <AuthProvider>
          <Component />
        </AuthProvider>
      );
    }
  };
};

export const renderWithNavigation = ({ screens = {} }) => {
  const NavigationComponent = withNavigation({ screens });
  const Component = withProviders(NavigationComponent);
  return { ...render(<Component />) };
};

export const updateFormWith = ({ values, getByLabelText }) => {
  const formInput = values;
  const elements = Object.keys(formInput).reduce((acc, name) => {
    acc[name] = getByLabelText(name);
    return acc;
  }, {});
  Object.keys(elements).forEach(item => {
    fireEvent.changeText(elements[item], formInput[item]);
  });
};
