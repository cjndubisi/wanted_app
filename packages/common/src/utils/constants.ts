// use 'react-native-config' & 'dotenv' for mobile and web respectively
import { Platform } from 'react-native';
interface IConfig {
  API_URL: string;
}

const nativeConfig = Platform.select({
  ios: () => require('react-native-config'),
  android: () => require('react-native-config'),
})();

const env = { ...process.env, ...nativeConfig.default };
const envKeys = Object.keys(env)
  .filter((item) => {
    return item.indexOf('REACT_APP_') !== -1;
  })
  .reduce((acc, next) => {
    console.log(acc, next, next.replace('REACT_APP_', ''));
    acc[next.replace('REACT_APP_', '')] = env[next];
    return acc;
  }, {});

// using ...process.env doesn't work,
// must manually access API_URL on .env
const Config: Partial<IConfig> = {
  ...envKeys,
};
export default Config;
