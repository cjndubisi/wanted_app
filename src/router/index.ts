export type RootStackParamList = {
  Splash: undefined;
  EmailAuth: undefined;
};
