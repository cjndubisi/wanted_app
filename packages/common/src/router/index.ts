import { AuthResolver, Splash, Home, EmailAuth } from '../screens';

export * as Stack from './stackRouter';

export const Routes = {
  AuthResolver: {
    path: '/authResolver',
    component: AuthResolver,
  },
  Splash: {
    path: '/',
    component: Splash,
  },
  Email: {
    path: '/authenticating',
    component: EmailAuth,
  },
  Home: {
    path: '/board',
    component: Home,
  },
};

export type RootStackParamList = {
  [P in keyof typeof Routes]: undefined;
};
