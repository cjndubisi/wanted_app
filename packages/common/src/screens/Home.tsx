import { StackNavigationProp } from '@react-navigation/stack';
import React, { useLayoutEffect } from 'react';
import { RootStackParamList } from '../router';
import { Container } from '../styled';

type SpashNavigationProp = StackNavigationProp<RootStackParamList, 'Home'>;
type Prop = {
  navigation: SpashNavigationProp;
};

export default ({ navigation }: Prop) => {
  useLayoutEffect(() => {
    navigation.setOptions({ headerShown: false });
  }, [navigation]);
  return <Container style={{ margin: 20 }}></Container>;
};
