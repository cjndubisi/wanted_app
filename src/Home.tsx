import { StackNavigationProp } from '@react-navigation/stack';
import React from 'react';
import { RootStackParamList } from './router';
import { Container } from './shared/styled';

type SpashNavigationProp = StackNavigationProp<RootStackParamList, 'Home'>;
type Prop = {
  navigation: SpashNavigationProp;
};

export default ({ navigation }: Prop) => {
  navigation.setOptions({ headerShown: false });

  return <Container style={{ margin: 20 }}></Container>;
};
