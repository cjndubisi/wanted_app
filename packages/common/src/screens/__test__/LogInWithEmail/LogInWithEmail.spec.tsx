jest.mock('node-fetch');

import { fireEvent } from '@testing-library/react-native';
import React from 'react';
import { create } from 'react-test-renderer';
import EmailLogin, { FormState } from '../../LogInWithEmail';
import { render, updateFormWith, withStackNavigation } from "../utils";
import { default as fetcher } from 'node-fetch';
import { Routes } from '../../../router';
import Home from '../../Home';
const fetch = (fetcher as any) as jest.Mock;
const { Response } = jest.requireActual('node-fetch');
const components = {
  screens: {
    EmailLogin: { component: EmailLogin, path: Routes.EmailLogin.path },
    Home: { component: Home, path: Routes.Home.path },
  },
};

it('renders correctly', () => {
  expect(render(withStackNavigation(components)).container).toMatchSnapshot();
});

test('cannot submit with empty fields', async () => {
  const { findByLabelText, getByText } = render(withStackNavigation(components));

  fireEvent.press(getByText(/Log in/i));

  await expect(findByLabelText('Last name is too short')).toBeTruthy();
});

test('test can sign up', async () => {
  const body = {
    user: {
      id: 1,
      first_name: 'random',
      last_name: 'random',
      email: 'random@random.com'
    },
    auth_token: 'faslfuad_random_fasd',
    message: 'success'
  };
  fetch.mockReturnValue(Promise.resolve(new Response(JSON.stringify(body))));

  const formInput: FormState = {
    email: 'random@random.com',
    password: 'R#and0m',
  };
  const { getByLabelText, getByText, findByText } = render(withStackNavigation(components));

  updateFormWith({ values: formInput, getByLabelText });

  fireEvent.press(getByText(/Log in/i));

  await expect(findByText(/Board/i)).toBeTruthy();
});
