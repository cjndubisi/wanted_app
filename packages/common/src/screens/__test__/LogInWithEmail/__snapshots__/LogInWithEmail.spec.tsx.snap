// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 0.01,
                    "restSpeedThreshold": 0.01,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 0.01,
                    "restSpeedThreshold": 0.01,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 64,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <SafeAreaView
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#fff7e6",
                              "maxWidth": 467,
                              "minHeight": "100%",
                              "paddingBottom": 20,
                              "paddingLeft": 20,
                              "paddingRight": 20,
                              "paddingTop": 20,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Object {
                              "margin": 20,
                              "marginTop": 44,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "marginBottom": 20,
                                "marginTop": 20,
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "fontSize": 24,
                                    "fontWeight": "bold",
                                    "marginBottom": 10,
                                  },
                                ]
                              }
                            >
                              Wanted
                            </Text>
                            <Text
                              style={
                                Array [
                                  Object {},
                                  Object {
                                    "color": "black",
                                    "marginBottom": 4,
                                  },
                                ]
                              }
                            >
                              Create an account buy and sell services, product, jobs and more.
                            </Text>
                          </View>
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "red",
                                  "fontSize": 12,
                                  "height": 13,
                                  "marginBottom": 6,
                                },
                              ]
                            }
                          />
                          <View
                            style={
                              Object {
                                "marginBottom": 12,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "display": "flex",
                                  "flex": 1,
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                }
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {},
                                    Object {
                                      "color": "black",
                                      "fontSize": 12,
                                      "fontWeight": "bold",
                                      "marginBottom": 4,
                                    },
                                  ]
                                }
                              >
                                Email
                              </Text>
                              <Text
                                accessibilityLabel="error_email"
                                style={
                                  Array [
                                    Object {
                                      "color": "red",
                                      "fontSize": 12,
                                      "height": 13,
                                      "marginBottom": 6,
                                    },
                                  ]
                                }
                              />
                            </View>
                            <TextInput
                              accessibilityLabel="email"
                              allowFontScaling={true}
                              rejectResponderTermination={true}
                              secureTextEntry={false}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "white",
                                    "borderRadius": 5,
                                    "color": "black",
                                    "fontSize": 14,
                                    "height": 40,
                                    "paddingLeft": 10,
                                    "paddingRight": 10,
                                  },
                                ]
                              }
                              underlineColorAndroid="transparent"
                              value=""
                            />
                          </View>
                          <View
                            style={
                              Object {
                                "marginBottom": 12,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "display": "flex",
                                  "flex": 1,
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                }
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {},
                                    Object {
                                      "color": "black",
                                      "fontSize": 12,
                                      "fontWeight": "bold",
                                      "marginBottom": 4,
                                    },
                                  ]
                                }
                              >
                                Password
                              </Text>
                              <Text
                                accessibilityLabel="error_password"
                                style={
                                  Array [
                                    Object {
                                      "color": "red",
                                      "fontSize": 12,
                                      "height": 13,
                                      "marginBottom": 6,
                                    },
                                  ]
                                }
                              />
                            </View>
                            <TextInput
                              accessibilityLabel="password"
                              allowFontScaling={true}
                              rejectResponderTermination={true}
                              secureTextEntry={true}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "white",
                                    "borderRadius": 5,
                                    "color": "black",
                                    "fontSize": 14,
                                    "height": 40,
                                    "paddingLeft": 10,
                                    "paddingRight": 10,
                                  },
                                ]
                              }
                              underlineColorAndroid="transparent"
                              value=""
                            />
                          </View>
                          <TouchableOpacity
                            activeOpacity={0.8}
                            backgroundColor="brown"
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "brown",
                                  "borderRadius": 5,
                                  "height": 40,
                                  "justifyContent": "center",
                                  "marginTop": 10,
                                },
                              ]
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {},
                                  Object {
                                    "color": "black",
                                    "fontWeight": "bold",
                                    "marginBottom": 4,
                                  },
                                  Object {
                                    "color": "white",
                                  },
                                ]
                              }
                            >
                              Log in
                            </Text>
                          </TouchableOpacity>
                        </View>
                      </SafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          pointerEvents="box-none"
          style={null}
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(224, 224, 224)",
                  "flex": 1,
                  "shadowColor": "rgb(224, 224, 224)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "height": 64,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 20,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "marginHorizontal": 18,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  /login
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
</View>
`;
